SSA/Projects/InstCombine/PaperExamples.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Released under Apache 2.0 license as described in the file LICENSE.
+-/
+import SSA.Projects.InstCombine.LLVM.PrettyEDSL
+import SSA.Projects.InstCombine.Refinement
+import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.Base
+import SSA.Projects.InstCombine.ForStd
+import SSA.Projects.InstCombine.ForMathlib
+import SSA.Core.ErasedContext
+import Batteries.Data.BitVec
+import Lean
+
+open MLIR AST
+open Std (BitVec)
+open Ctxt (Var DerivedCtxt)
+open InstCombine (MOp)
+open Lean
+
+namespace AliveHandwritten
+set_option pp.proofs false
+set_option pp.proofs.withType false
+
+
+theorem xor_allOnes_eq_not (x: _root_.BitVec w) :
+  x ^^^ (BitVec.allOnes w) = ~~~ x:= by
+  ext i
+  simp
+
+-- example proof of pure bitvector reasoning.
+-- This is closed automatically by proof automation.
+example (w : Nat) :
+  [llvm ( w )| {
+  ^bb0(%X : _, %C1 : _, %C2 : _):
+    %v1 = llvm.xor %X, %C1
+    %v2 = llvm.and %v1, %C2
+    llvm.return %v2
+  }] ⊑ [llvm ( w )| {
+  ^bb0(%X : _, %C1 : _, %C2 : _):
+  %v1 = llvm.and %X, %C2
+  %v2 = llvm.xor %X, %C1
+  %v3 = llvm.and %C1, %C2
+  %v4 = llvm.xor %v1, %v3
+  llvm.return %v4
+  }] := by simp_alive_peephole; alive_auto
+
+-- example proof of add plus xor.
+-- This is an easy alive statement.
+example (w : Nat) :
```

Comment:
this example is not going to the paper, right? should we remove it from here?

Reply:
Deleted.